spring-rabbit/src/main/java/org/springframework/amqp/rabbit/logback/AmqpAppender.java

```diff
@@ -435,8 +556,60 @@ public void start() {
 	}
 
 	/**
+	 * Create the {@link ConnectionFactory}.
+	 *
+	 * @return a {@link ConnectionFactory}.
+	 */
+	protected ConnectionFactory createRabbitConnectionFactory() {
+		RabbitConnectionFactoryBean factoryBean = new RabbitConnectionFactoryBean();
+		configureRabbitConnectionFactory(factoryBean);
+		try {
+			factoryBean.afterPropertiesSet();
+			return factoryBean.getObject();
+		}
+		catch (Exception e) {
+			addError("Failed to create customized Rabbit ConnectionFactory.", e);
+			return null;
```

Comment:
The reason why I chose to return `null` and hence passing `null` to the constructor of the `CachingConnectionFactory` was to simply cause a `RuntimeException` that would stop the application from starting. I agree it would be better to avoid the second error message in this case (the `keyStore property required` message is enough). 

What would you propose the logic if the `if..else` statement in the `start()` method do? Would it just not proceed with setting up the appender while allowing the application to start? Or throw a variant of `RuntimeException` to cause the application to fail?